assets/js/modules/analytics/datastore/profiles.js

```diff
@@ -0,0 +1,342 @@
+	 * @param {string} args.propertyId Google Analytics property ID.
+	 * @param {Object} args.error Error object.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveCreateProfileFailed( { accountId, error, propertyId } ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( error, 'error is required.' );
+		invariant( propertyId, 'propertyId is required.' );
+
+		return {
+			payload: { accountId, error, propertyId },
+			type: RECEIVE_CREATE_PROFILE_FAILED,
+		};
+	},
+
+	receiveProfiles( { accountId, propertyId, profiles } ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( propertyId, 'accountId is required' );
+		invariant( profiles, 'profiles is required' );
+
+		return {
+			payload: { accountId, propertyId, profiles },
+			type: RECEIVE_PROFILES,
+		};
+	},
+
+	receiveProfilesFailed( { accountId, error, propertyId } ) {
+		invariant( accountId, 'accountId is required' );
+		invariant( error, 'error is required.' );
+		invariant( propertyId, 'accountId is required' );
+
+		return {
+			payload: { accountId, error, propertyId },
+			type: RECEIVE_PROFILES_FAILED,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_CREATE_PROFILE ]: ( accountId, propertyId ) => {
+		return API.set( 'modules', 'analytics', 'create-profile', {
+			accountID: accountId,
+			propertyID: propertyId,
+		} );
+	},
+	[ FETCH_PROFILES ]: ( accountId, propertyId ) => {
+		return API.get( 'modules', 'analytics', 'profiles', {
+			accountID: accountId,
+			propertyID: propertyId,
+		} );
+	},
+};
+
+export const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case FETCH_CREATE_PROFILE: {
+			const { accountId, propertyId } = action.payload;
+
+			return {
+				...state,
+				isDoingCreateProfile: {
+					...state.isDoingCreateProfile,
+					[ `${ accountId }::${ propertyId }` ]: true,
+				},
+			};
+		}
+
+		case FETCH_PROFILES: {
+			const { accountId, propertyId } = action.payload;
+
+			return {
+				...state,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountId }::${ propertyId }` ]: true,
+				},
+			};
+		}
+
+		case RECEIVE_CREATE_PROFILE: {
+			const { accountId, propertyId, profile } = action.payload;
+
+			return {
+				...state,
+				isDoingCreateProfile: {
+					...state.isDoingCreateProfile,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+				profiles: [
+					...state.profiles || [],
+					profile,
+				],
+			};
+		}
+
+		case RECEIVE_CREATE_PROFILE_FAILED: {
+			const { accountId, error, propertyId } = action.payload;
+
+			return {
+				...state,
+				error,
+				isDoingCreateProfile: {
+					...state.isDoingCreateProfile,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+			};
+		}
+
+		case RECEIVE_PROFILES: {
+			const { accountId, propertyId, profiles } = action.payload;
+
+			return {
+				...state,
+				isFetchingProfiles: {
+					...state.isFetchingProfiles,
+					[ `${ accountId }::${ propertyId }` ]: false,
+				},
+				profiles,
```

Comment:
This should not be passed through (see other comments), but integrated into `state.profiles` which should be keyed by account ID on the top level and by property ID on the second level.